feat(speech): add listen and speak creation operators

kosich opened the pull request:
Hey! 👋

This is a migration of [rxjs-tts](https://github.com/kosich/rxjs-tts) and [rxjs-stt](https://github.com/kosich/rxjs-stt) libraries

This PR adds support for [Web Speech API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Speech_API):

- `fromSpeechRecognition` creation operator that wraps [SpeechRecognition](https://developer.mozilla.org/en-US/docs/Web/API/SpeechRecognition)
  _originally this method was called `listen`_

- and `speak` creation operator that wraps [SpeechSynthesis](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesis)
  _couldn't come up with a better name: `fromSpeechSynthesis` doesn't sound exactly right since it's rather a generation, not subscribing for events_

Both have dedicated README-s with basic API examples. I suggest either including those in the root README, or linking child readmes from the root.

Please, review 🙂
Thanks

kosich committed:
feat(speech): add listen and speak creation operators

kosich commented:
Had to update prettier config to match linter: drop parenthesis on arrow functions
And set `.md` to be ignored by prettier — as we often need custom formatting in code samples we give in READMEs

kosich commented:
The build is still failing on runners with node v `< 12` due to `globalThis` usage (which is kinda expected as only 12+ versions support it)

Why do we need tests run on different node versions: IMO, one latest version should be enough since the lib is targeted on web platforms rather then node?

@niklas-wortmann please, advise what to do with failing build.

niklas-wortmann commented:
Hi sorry for the late reply, thanks for your contribution. I really appreciate it. To be honest, I am poorly prepared for contributors right now, so the experience kinda sucks. I am really sorry about this. Please don't mind the failing pipeline I will take care of this.

niklas-wortmann commented on libs/rxjs-web/src/lib/web-api/SpeechAPI/SpeechRecognition/fromSpeechRecognition.ts:
```diff
@@ -0,0 +1,94 @@
+		return subscription;
+	});
+
+	return timer(4).pipe(mergeMapTo(recognition$));
```
could you elaborate on the `timer(4)`

kosich replied:
Fair point! I don't remember 😅

There were some FF and cancellation issues, not sure what was exact reason for this. Will have to investigate.

niklas-wortmann reviewed (approved):
Looks amazing, I am just interested in the timer thingy, is it just to give an extra pause?

kosich commented:
Can you also help with naming?

I feel like both either should be `listen`/`speak` or `fromSpeechRecognition`/`fromSpeechSynthesis`. What do you think?

niklas-wortmann commented:
I actually like listen and speak, let's start with that and see how it goes

niklas-wortmann commented:
@kosich could you do me the favor and rename the operators as discussed, when you pushed those changes, I would be extremely happy to merge this pr.

kosich commented:
Hey, Jan-Niklas! I'm sorry about this delay. I was procrastinating and postponing testing that FF delay.
I've tried some scenarios on FF and Chrome and it seems that 4ms delay was useless so I removed it.
Not sure why I added it in the first place 🤷‍♂️

And I've renamed the speech-to-text fn back to `listen`.

CI on node 13 & 12 passes, node 11 & 10 fails (as expected).
The PR is ready for merge

Please, ping me if I missed something 🙂

niklas-wortmann commented:
Whooop woop thanks a lot for your work! I appreciate it a lot!

kosich commented:
🙌 Glad to collaborate 🙂

I will deprecate my packages in favor of `rxjs-web`

Now that this is merged, I'd also suggest redesigning `README.md` for better library consumer landing / docs navigation.
_(well, and to make it sexy)_
